contracts/GenericLender/GenericScream.sol

```diff
@@ -0,0 +1,308 @@
+    }
+
+    //WARNING. manipulatable and simple routing. Only use for safe functions
+    function priceCheck(address start, address end, uint256 _amount) public view returns (uint256) {
+        if (_amount == 0) {
+            return 0;
+        }
+        address[] memory path = getTokenOutPath(start, end);
+        uint256[] memory amounts = IUniswapV2Router02(spookyRouter).getAmountsOut(_amount, path);
+
+        return amounts[amounts.length - 1];
+    }
+
+    function weightedApr() external view override returns (uint256) {
+        uint256 a = _apr();
+        return a.mul(_nav());
+    }
+
+    function withdraw(uint256 amount) external override management returns (uint256) {
+        return _withdraw(amount);
+    }
+
+    //emergency withdraw. sends balance plus amount to governance
+    function emergencyWithdraw(uint256 amount) external override management {
+        //dont care about errors here. we want to exit what we can
+        cToken.redeem(amount);
+
+        want.safeTransfer(vault.governance(), want.balanceOf(address(this)));
+    }
+
+    //withdraw an amount including any want balance
+    function _withdraw(uint256 amount) internal returns (uint256) {
+        uint256 balanceUnderlying = cToken.balanceOfUnderlying(address(this));
+        uint256 looseBalance = want.balanceOf(address(this));
+        uint256 total = balanceUnderlying.add(looseBalance);
+
+        if (amount.add(dustThreshold) >= total) {
+            //cant withdraw more than we own. so withdraw all we can
+            if(balanceUnderlying > dustThreshold){
+                require(cToken.redeem(cToken.balanceOf(address(this))) == 0, "ctoken: redeemAll fail");
+            }
+            looseBalance = want.balanceOf(address(this));
+            if(looseBalance > 0 ){
+                want.safeTransfer(address(strategy), looseBalance);
+                return looseBalance;
+            }else{
+                return 0;
+            }
+
+        }
+
+        if (looseBalance >= amount) {
+            want.safeTransfer(address(strategy), amount);
+            return amount;
+        }
+
+        //not state changing but OK because of previous call
+        uint256 liquidity = want.balanceOf(address(cToken));
+
+        if (liquidity > 1) {
+            uint256 toWithdraw = amount.sub(looseBalance);
+
+            if (toWithdraw > liquidity) {
+                toWithdraw = liquidity;
+            }
+            if(toWithdraw > dustThreshold){
+                require(cToken.redeemUnderlying(toWithdraw) == 0, "ctoken: redeemUnderlying fail");
+            }
+
+        }
+        _disposeOfComp();
+        looseBalance = want.balanceOf(address(this));
+        want.safeTransfer(address(strategy), looseBalance);
+        return looseBalance;
+    }
+
+    function _disposeOfComp() internal {
+
+        CTokenI[] memory tokens = new CTokenI[](1);
+        tokens[0] = cToken;
+
+        ComptrollerI(unitroller).claimComp(address(this), tokens);
```

Comment:
taking into account the current Comp situation, do we want to be able to disable the claim of comp? otherwise it might break the withdraw function

Reply:
Add a setter? `if (claimComp)`then we claim?

Reply:
done